pkg/querier/codec/protobuf_codec.go

```diff
@@ -13,10 +13,20 @@ import (
 	"github.com/cortexproject/cortex/pkg/querier/tripperware"
 )
 
-type ProtobufCodec struct{}
+type ProtobufCodec struct {
+	// cortexInternal enables encoding the whole native histogram data fields in response instead of keeping
+	// only few sparse information like the default JSON/Protobuf codec does.
+	// This will be used by Cortex Ruler to get native histograms data from Cortex Query Frontend because
+	// rule evaluation requires the full native histogram data.
+	CortexInternal bool
+}
 
 func (p ProtobufCodec) ContentType() v1.MIMEType {
-	return v1.MIMEType{Type: "application", SubType: "x-protobuf"}
+	if !p.CortexInternal {
+		return v1.MIMEType{Type: "application", SubType: "x-protobuf"}
+	}
+	// TODO: switch to use constants.
+	return v1.MIMEType{Type: "application", SubType: "application/x-cortex-query+proto"}
```

Comment:
subType is a `x-cortex-query+proto`
```suggestion
	eturn v1.MIMEType{Type: "application", SubType: "x-cortex-query+proto"}
```